### ISSUE TITLE
mate applets crash with new cairo version introduced on the Arch Linux core repos

### ISSUE THREAD

**Issue by egnappahz** [mate-desktop/mate-applets#556]
Hello, after updating cairo on ArchLinx to version "1.17.4", I have the following issue:

Steps to reproduce:
Use workspace switcher or active window list buttons applets in mate

when pointed at it with the mouse cursor, the applet crash with:

**org.mate.panel.applet.WnckletFactory[10494]: wnck-applet: ../cairo/src/cairo-surface.c:1733: cairo_surface_mark_dirty_rectangle: Assertion `! _cairo_surface_has_snapshots (surface)' failed.**

Reloading applets has no effect. Crash is almost instantly with just moving the mouse over the affected applets.

Downgrading cairo solves this issue:
[2020-12-04T08:36:27+0100] [ALPM] downgraded cairo (1.17.4-11.17.4-1 -> 1.17.2+25+gaee96d175-1)
[2020-12-04T08:36:27+0100] [ALPM] downgraded lib32-cairo (1.17.4-1 -> 1.17.2+25+gaee96d175-1)

I have this happen on 3 seperate physical Arch linux machines, which makes it hard for me to believe it was just a one-off

Now, to investigate I even tried to compile mate from source since I had done this in the past and the exact same error still persists (when installing them on a fresh system). These are my compiled packages on a seperate, UP TO DATE (so the latest cairo version) arch linux system (I used the tagged versions of your git, see filenames:
----
atril-git-1.24.0-1-x86_64.pkg.tar.zst
caja-extensions-git-1.24.1-1-x86_64.pkg.tar.zst
caja-git-1.24.0-1-x86_64.pkg.tar.zst
engrampa-git-1.24.1-1-x86_64.pkg.tar.zst
eom-git-1.24.1-1-x86_64.pkg.tar.zst
libmatekbd-git-1.24.1-1-x86_64.pkg.tar.zst
libmatemixer-git-1.24.1-1-x86_64.pkg.tar.zst
libmateweather-git-1.24.1-1-x86_64.pkg.tar.zst
marco-git-1.24.1-1-x86_64.pkg.tar.zst
mate-applets-git-1.24.1-1-x86_64.pkg.tar.zst
mate-backgrounds-git-1.24.1-1-x86_64.pkg.tar.zst
mate-calc-git-eggz-1.24.1-1-x86_64.pkg.tar.zst
mate-common-git-1.24.1-1-x86_64.pkg.tar.zst
mate-control-center-git-1.24.1-1-x86_64.pkg.tar.zst
mate-desktop-git-1.24.1-1-x86_64.pkg.tar.zst
mate-icon-theme-git-1.24.0-1-x86_64.pkg.tar.zst
mate-media-git-1.24.1-1-x86_64.pkg.tar.zst
mate-menus-git-1.24.1-1-x86_64.pkg.tar.zst
mate-notification-daemon-git-1.24.1-1-x86_64.pkg.tar.zst
mate-panel-git-1.24.1-1-x86_64.pkg.tar.zst
mate-polkit-git-1.24.0-1-x86_64.pkg.tar.zst
mate-power-manager-git-1.24.1-1-x86_64.pkg.tar.zst
mate-screensaver-git-1.24.1-1-x86_64.pkg.tar.zst
mate-sensors-applet-git-1.24.1-1-x86_64.pkg.tar.zst
mate-session-manager-git-1.24.1-1-x86_64.pkg.tar.zst
mate-settings-daemon-git-1.24.1-1-x86_64.pkg.tar.zst
mate-system-monitor-git-1.24.1-1-x86_64.pkg.tar.zst
mate-terminal-git-1.24.1-1-x86_64.pkg.tar.zst
mate-themes-git-3.22.9-1-x86_64.pkg.tar.zst
mate-utils-git-1.24.0-1-x86_64.pkg.tar.zst
mozo-git-1.24.0-1-x86_64.pkg.tar.zst
pluma-git-1.24.1-1-x86_64.pkg.tar.zst
python-caja-git-1.24.0-1-x86_64.pkg.tar.zst
----

TL;DR
Do you guys have any idea why the new cairo version throws this error on your mate DE?

org.mate.panel.applet.WnckletFactory[10494]: wnck-applet: ../cairo/src/cairo-surface.c:1733: cairo_surface_mark_dirty_rectangle: Assertion `! _cairo_surface_has_snapshots (surface)' failed.

**Comment by vkareh** [comment]
This issue belongs in mate-panel and is a duplicate of https://github.com/mate-desktop/mate-panel/issues/1166
A possible fix is submitted here: https://github.com/mate-desktop/mate-panel/pull/1168

Closing this
